driver-core/src/main/com/mongodb/connection/DefaultAuthenticator.java

```diff
@@ -16,33 +16,95 @@
 
 package com.mongodb.connection;
 
+import com.mongodb.AuthenticationMechanism;
 import com.mongodb.MongoCredential;
+import com.mongodb.MongoSecurityException;
 import com.mongodb.async.SingleResultCallback;
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonInt32;
+import org.bson.BsonString;
 
+import static com.mongodb.AuthenticationMechanism.SCRAM_SHA_1;
+import static com.mongodb.AuthenticationMechanism.SCRAM_SHA_256;
 import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.connection.CommandHelper.executeCommand;
+import static com.mongodb.connection.CommandHelper.executeCommandAsync;
+import static java.lang.String.format;
+import static java.util.Collections.singletonList;
 
 class DefaultAuthenticator extends Authenticator {
+    private static final ServerVersion FOUR_ZERO = new ServerVersion(3, 7);
+    private static final ServerVersion THREE_ZERO = new ServerVersion(3, 0);
+    private static final BsonString DEFAULT_MECHANISM_NAME = new BsonString(SCRAM_SHA_256.getMechanismName());
+    private static final BsonArray DEFAULT_SUPPORTED_MECHANISMS = new BsonArray(singletonList(DEFAULT_MECHANISM_NAME));
```

Comment:
The proposed spec says:

    If the command succeeds and the response does not include a ``saslSupportedMechs`` field, then drivers MUST use the legacy default mechanism rules for servers older than 4.0

So I think this should contain SCRAM-SHA-1 not SCRAM-SHA-256.